mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -2022,6 +2025,100 @@ func TestClientSideEncryptionProse(t *testing.T) {
 		})
 	})
 
+	mt.RunOpts("18. Azure IMDS Credentials", noClientOpts, func(mt *mtest.T) {
+		buf := make([]byte, 0, 256)
+		kmsProvidersMap := map[string]map[string]interface{}{
+			"azure": {},
+		}
+		p, err := bson.MarshalAppendWithRegistry(bson.DefaultRegistry, buf[:0], kmsProvidersMap)
```

Comment:
Optional: `MarshalAppendWithRegistry` will be removed in Go Driver 2.0. Since you're using `bson.DefaultRegistry`, consider using `bson.MarshalAppend` instead.

```suggestion
		p, err := bson.MarshalAppend(buf[:0], kmsProvidersMap)
```